recipes-bsp/u-boot/u-boot-iot2050.inc

```diff
@@ -20,7 +20,7 @@ U_BOOT_CONFIG = "iot2050_defconfig"
 U_BOOT_BIN = "flash.bin"
 
 SPI_FLASH_IMG = "${U_BOOT_BIN}"
-SPI_FLASH_DEPLOY_IMG = "iot2050-image-boot.bin"
+SPI_FLASH_DEPLOY_IMG = "iot2050-${IOT2050_VARIANTS}-image-boot.bin"
```

Comment:
Set this var in the including recipe, then `IOT2050_VARIANTS` is not needed also for this use case.